keras_cv/layers/preprocessing/random_rotation_test.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+import tensorflow as tf
+
+from keras_cv.layers.preprocessing.random_rotation import RandomRotation
+
+
+class RandomRotationTest(tf.test.TestCase):
+    def test_random_rotation_output_shapes(self):
+        input_images = np.random.random((2, 5, 8, 3)).astype(np.float32)
+        expected_output = input_images
+        layer = RandomRotation(0.5)
+        actual_output = layer(input_images, training=True)
+        self.assertEqual(expected_output.shape, actual_output.shape)
+
+    def test_random_rotation_inference(self):
+        input_images = np.random.random((2, 5, 8, 3)).astype(np.float32)
+        expected_output = input_images
+        layer = RandomRotation(0.5)
+        actual_output = layer(input_images, training=False)
+        self.assertAllClose(expected_output, actual_output)
+
+    def test_distribution_strategy(self):
+        """Tests that RandomRotation can be created within distribution
+        strategies."""
+        input_images = np.random.random((2, 5, 8, 3)).astype(np.float32)
+        strat = tf.distribute.MirroredStrategy(devices=["cpu", "gpu"])
+        with strat.scope():
+            layer = RandomRotation(0.5)
+            output = strat.run(lambda: layer(input_images, training=True))
+        values = output.values
+        self.assertAllEqual(2, len(values))
+
+    def test_config_with_custom_name(self):
+        layer = RandomRotation(0.5, name="image_preproc")
+        config = layer.get_config()
+        layer_1 = RandomRotation.from_config(config)
```

Comment:
any reason for layer_1 here? I didn't see layer_2.

Reply:
there is layer and layer_1

Reply:
I think layer_reconstructed would be a good name

Reply:
Done